tests/FINDOLOGIC/Tests/PropertyTest.php

```diff
@@ -27,4 +27,28 @@ public function testAddingMultipleValuesWithoutUsergroupCausesException()
         $property->addValue('foobar');
         $property->addValue('foobar');
     }
+
+    /**
+     * @expectedException \FINDOLOGIC\Export\Data\PropertyKeyNotAllowedException
+     */
+    public function testReservedPropertyKeyOrdernumberCausesException()
```

Comment:
Repetitive tests can be avoided by using [data providers](https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers).

Reply:
Example (didn't try it yet):

```php
public function propertyKeyProvider()
{
    return array(
        'reserved property "ordernumber"' => array('ordernumber', true),
        'reserved property "thumbnail\d+"' => array('image0', true),
        'reserved property "image\d+"' => array('thumbnail1', true),
        'non-reserved property key' => array('foobar', true)
    );
}

/**
  * @dataProvider propertyKeyProvider
  */
public function testReservedPropertyKeysCauseException($key, $shouldCauseException) {
{
    try {
        new Property($key);
        if ($shouldCauseException) {
            $this->fail('Using a reserved property key should cause an exception.');
        }
    } catch ($e) {
        if ($shouldCauseException) {
            $this->assertRegexp('/' . $key . '/', $e->getMessage());
        }
    }
}
```

Reply:
Done